android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java

```diff
@@ -135,6 +137,25 @@ public void onBindViewHolder(final SessionViewHolder holder, final int position)
             holder.sessionSubtitle.setVisibility(View.VISIBLE);
             holder.sessionSubtitle.setText(sessionSubTitle);
         }
+        holder.sessionStatus.setVisibility(View.GONE);
+        DateUtils curr_date = new DateUtils();
+        Date start;
+        Date end;
+        try {
+            start = curr_date.getDate(session.getStartsAt());
+            end = curr_date.getDate((session.getEndsAt()));
+            Date current = new Date();
+            if(start.after(current)) {
+                holder.sessionStatus.setVisibility(View.VISIBLE);
+                holder.sessionStatus.setText("UPCOMING");
+            }
+            else if(((start.before(current)) || (start.equals(current))) && ((end.after(current)) || (end.equals(current)))){
```

Comment:
Move else if in above line and if this condition is same as used above, create a function returning a boolean for it